tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_provision.c

```diff
@@ -172,6 +205,45 @@ static void test_provisioner_pb_adv_multi(void)
 	PASS();
 }
 
+static void test_provisioner_pb_adv_reprovision(void)
+{
+	int err;
+
+	k_sem_init(&prov_sem, 0, 1);
+
+	bt_mesh_device_setup();
+
+	err = bt_mesh_cdb_create(test_net_key);
+	ASSERT_OK(err, "Failed to create CDB (err %d)\n", err);
+
+	err = bt_mesh_provision(test_net_key, 0, 0, 0, 0x0001, dev_key);
+	ASSERT_OK(err, "Provisioning failed (err %d)", err);
+
+	for (int i = 0; i < PROV_REPROV_COUNT; i++) {
+		LOG_INF("Prov'ner prov loop #%d, waiting for prov ...\n", i);
+		ASSERT_OK(k_sem_take(&prov_sem, K_SECONDS(20)),
+			"Provisioner reprovision #%d fail", i);
+
+		err = bt_mesh_cfg_mod_sub_add(0, prov_addr - 1, 0x0000, 0xc000, BT_MESH_MODEL_ID_HEALTH_SRV, NULL);
```

Comment:
This and the other parts could probably benefit from a comment or two, as the configuration changes look kind of pointless on the surface

Reply:
minor: `prov_addr - 1` could be called as `current_dev_addr` and set in the node addition callback above.